WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModel.kt

```diff
@@ -52,10 +65,37 @@ class EditPostPublishSettingsViewModel @Inject constructor(
         if (isStarted) return
         isStarted = true
 
-        postRepository?.let {
-            val site = siteStore.getSiteByLocalId(it.localSiteId) ?: return@let
-            if (site.hasCapabilityListUsers) {
-                fetchAuthors(site)
+        siteModel = postRepository?.localSiteId?.let {
+            siteStore.getSiteByLocalId(it)
+        }
+        loadAuthors()
+        loadJetpackSocial()
+    }
+
+    private fun loadAuthors() {
+        siteModel?.let {
+            if (it.hasCapabilityListUsers) {
+                fetchAuthors(it)
+            }
+        }
+    }
+
+    private fun loadJetpackSocial() {
+        if (!jetpackSocialFeatureConfig.isEnabled()) {
+            _showJetpackSocialContainer.value = false
+            return
+        }
+        siteModel?.let {
+            _showJetpackSocialContainer.value = true
+            viewModelScope.launch {
+                val connections = getPublicizeConnectionsForUserUseCase.execute(it.siteId, accountStore.account.userId)
+                val shareLimit = getJetpackSocialShareLimitStatusUseCase.execute(it)
+                val state = if (connections.isEmpty()) {
+                    jetpackUiStateMapper.mapNoConnections(::onJetpackSocialConnectProfilesClick)
+                } else {
+                    jetpackUiStateMapper.mapLoaded(connections, shareLimit, ::onJetpackSocialSubscribeClick)
+                }
+                _jetpackSocialUiState.postValue(state)
```

Comment:
Good point! In this case, the Jetpack Social container will keep its default visibility from XML (which is `GONE`) but I agree it would be better to have it explicitly set here in case someone inadvertently makes it visible. I'll make sure we call `_showJetpackSocialContainer.value = false` if `siteModel` is `null`.